src/components/Reports/useSDGProgress.ts

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react';
+import useDebounce from 'src/utils/useDebounce';
+import { Report, SDG } from 'src/types/Report';
+
+const DEBOUNCE_TIME_MS = 500;
+
+export default function useSDGProgress(
+  report: Report,
+  updateSDGProgress?: (index: number, value: string) => void
+): [string[], React.Dispatch<React.SetStateAction<string>>[]] {
+  const sdgProgressStates: string[] = [];
+  const setSdgProgressStates: React.Dispatch<React.SetStateAction<string>>[] = [];
+
+  // 1. NoPoverty
+  const [sdgProgress0, setSdgProgress0] = useState(
+    report.annualDetails?.sustainableDevelopmentGoals?.find((s) => s.goal === SDG[0])?.progress ?? ''
+  );
+  sdgProgressStates.push(sdgProgress0);
+  setSdgProgressStates.push(setSdgProgress0);
+  useDebounce(sdgProgress0, DEBOUNCE_TIME_MS, (value) => {
+    const index = report.annualDetails?.sustainableDevelopmentGoals?.findIndex((s) => s.goal === SDG[0]);
+    if (updateSDGProgress && index !== undefined && index >= 0) {
+      updateSDGProgress(index, value);
+    }
+  });
```

Comment:
if it helps.. you can extract this out into another hook... and call that hook sequentially, passing in the sdgProgressStates array and sdg index